### PR TITLE
Started roadmap, updated OS request guide for EL9 transition

### PR DESCRIPTION
**Pull request by rachellombardi**

### CODE REVIEW

**Comment by rachellombardi** [comment]
The roadmap isn't ready to be put on the website (and it won't be when this PR is merged - there is no header). But this will make it so that we can all edit the document as needed.

**Comment by rachellombardi** [comment]
Added separate transition guide

File: _uw-research-computing/htc-el8-to-el9-transition.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+* **April 2023**: HTC system will support CentOS 7, CentOS Stream 8, and CentOS Stream 9. By default, 
+all jobs not using a software container will continue to match to servers running CentOS 8, however,
+**users should begin using software containers or testing jobs on servers running CentOS Stream 9**. 
+* **May 2023**: Default operating system requirements for jobs will change from CentOS 8 to 
+CentOS Stream 9.
+
+## What You Need to Do
+* **No action is needed for researchers already using a Docker or Singularity/Apptainer software containers in their jobs.** Becuase software containers have a small operating system installed inside of them, these jobs carry everything they need with them and do not rely signifcantly on the host operating system. By default, your jobs will match to any operating system in the HTC pool, including the new CentOS Stream 9 hosts. 
+* **Researchers not already using a Docker or Apptainer software container will need to either (a) test their software/code on a CentOS Stream 9 machine to see their software needs to be reinstalled or (b) switch to using a software container (recommended). Option A **
```
Comment (rachellombardi) [inline _uw-research-computing/htc-el8-to-el9-transition.md:37]:
```suggestion
* **Researchers not already using a Docker or Apptainer software container will need to either (a) test their software/code on a CentOS Stream 9 machine to see their software needs to be reinstalled or (b) switch to using a software container (recommended). See the [below](#option-1-using-a-container-recommended) for additional information** 
```

File: _uw-research-computing/htc-el8-to-el9-transition.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+## What You Need to Do
+* **No action is needed for researchers already using a Docker or Singularity/Apptainer software containers in their jobs.** Becuase software containers have a small operating system installed inside of them, these jobs carry everything they need with them and do not rely signifcantly on the host operating system. By default, your jobs will match to any operating system in the HTC pool, including the new CentOS Stream 9 hosts. 
+* **Researchers not already using a Docker or Apptainer software container will need to either (a) test their software/code on a CentOS Stream 9 machine to see their software needs to be reinstalled or (b) switch to using a software container (recommended). Option A ** 
+
+## Troubleshooting and Additional Comments
+* **If your jobs were running successfully before, but now failing**, scale down your submissions 
+and run a few test jobs on the new operating system (now the default). If you absolutely 
+HAVE to keep running, you can continue to (temporarily) run on CentOS 8 by following 
+the instructions [below](#requesting-a-specific-operating-system)
```
Comment (ChristinaLK) [inline _uw-research-computing/htc-el8-to-el9-transition.md:43]:
```suggestion
## Troubleshooting and Additional Comments
```
This only makes sense after we flip the default.

File: _uw-research-computing/htc-el8-to-el9-transition.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+* **If you are having trouble getting your jobs to run successfully on the new operating system**, 
+please contact the facilitation team at chtc@cs.wisc.edu or [come to office hours](/uw-research-computing/get-help.html)
+* **If you would like to access as much computing capacity as possible**, consider using a Docker or Apptainer software container for your jobs so that your jobs can match to a variety of operating systems. See the [information below](#option-1-using-a-container-recommended) for detailed instructions on creating and using software containers. 
+
+## Use a Container or Request a Specific Operating System
```
Comment (ChristinaLK) [inline _uw-research-computing/htc-el8-to-el9-transition.md:48]:
```suggestion
## Options For Transitioning Your Jobs
```

File: _uw-research-computing/htc-el8-to-el9-transition.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+1. [Use a Container (recommended)](#option-1-using-a-container-recommended)
+1. [Request a Specific Operating System](#option-2-requesting-a-specific-operating-system).
+
+## Option 1: Use a Software Container (Recommended)
```
Comment (ChristinaLK) [inline _uw-research-computing/htc-el8-to-el9-transition.md:53]:
```suggestion
### Option 1: Use a Software Container (Recommended)
```

File: _uw-research-computing/htc-el8-to-el9-transition.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+- [Advanced Apptainer Example - SUMO](apptainer-htc-advanced-example.html)
+  
+**Docker**
+- [Explore and Test Docker Containers](docker-test.html)
+- [Build a Docker Container Image](docker-build.html)
+- [Running HTC Jobs Using Docker Containers](docker-jobs.html)
+
+**CHTC users are welcome to reach out to the Facilitation team via email or in office hours for help installing their software into a container.** 
+
+## Option 2: Request a Specific Operating System
```
Comment (ChristinaLK) [inline _uw-research-computing/htc-el8-to-el9-transition.md:72]:
```suggestion
### Option 2: Request a Specific Operating System
```

**Review by ChristinaLK — APPROVED** [review]
changing a few titles

**Comment by ChristinaLK** [comment]
Should probably still rename os-transition-guide (and then make sure the link from the el9 transition guide works) and remove the "requirements" version of requesting an operating system and just use what Theng has been working on.

File: _uw-research-computing/os-transition-htc.md [hunk 6]
```diff
@@ -40,31 +39,36 @@ version (or versions) for your jobs. This option is more limiting because
...
+### Use Both CentOS Stream 8 (previous default) and CentOS Stream 9 (current default)
 
 To request that your jobs run on computers running **either** version of 
 CentOS Linux, add the following requirements line to your submit file:
 
 ``` {.sub}
-requirements = (OpSysMajorVer == 7) || (OpSysMajorVer == 8)
+requirements = (OpSysMajorVer == 8) || (OpSysMajorVer == 9)
 ```
+If you are not comfortable with using class ads in your requirements statement, you can also add one or both of these options to your submit file instead:
```
Comment (rachellombardi) [inline _uw-research-computing/os-transition-htc.md:50]:
```suggestion
If you are not comfortable with using class ads in your requirements statement, you can also add **one** of these options to your submit file instead: 
```

File: _uw-research-computing/os-transition-htc.md [hunk 7]
```diff
@@ -40,31 +39,36 @@ version (or versions) for your jobs. This option is more limiting because
...
 
 ``` {.sub}
-requirements = (OpSysMajorVer == 7) || (OpSysMajorVer == 8)
+requirements = (OpSysMajorVer == 8) || (OpSysMajorVer == 9)
 ```
+If you are not comfortable with using class ads in your requirements statement, you can also add one or both of these options to your submit file instead: 
 
-> Note: this requirement is not necessary for jobs that use Docker containers; 
+```{.sub}
++WantEL9  # tells HTCondor your job can be matched to CentOS Stream 9 machines
```
Comment (rachellombardi) [inline _uw-research-computing/os-transition-htc.md:53]:
```suggestion
chtc_want_el9 = true  # tells HTCondor your job should be matched to CentOS Stream 9 machines
# or
```

File: _uw-research-computing/os-transition-htc.md [hunk 8]
```diff
@@ -40,31 +39,36 @@ version (or versions) for your jobs. This option is more limiting because
...
-### Require CentOS 7 (previous default)
+### Require CentOS 8 (previous default)
 
-To request that your jobs run on servers with CentOS 7 **only** add the
+To request that your jobs run on servers with CentOS 8 **only** add the
 following requirements line to your submit file:
 
 ``` {.sub}
-requirements = (OpSysMajorVer == 7)
+requirements = (OpSysMajorVer == 8)
```
Comment (rachellombardi) [inline _uw-research-computing/os-transition-htc.md:71]:
```suggestion
requirements = (OpSysMajorVer == 8)
# or 
chtc_want_el8 = true
```

File: _uw-research-computing/os-transition-htc.md [hunk 9]
```diff
@@ -81,7 +85,13 @@ requirements = (Target.HasCHTCStaging == true)
...
 
 ``` {.submit}
-requirements = (Target.HasCHTCStaging == true) && (OpSysMajorVer == 7)
+requirements = (Target.HasCHTCStaging == true) && (OpSysMajorVer == 8)
+```
+Or use the +WantEL syntax: 
+
+``` {.submit}
+requirements = (Target.HasCHTCStaging == true)
++WantEL8
```
Comment (rachellombardi) [inline _uw-research-computing/os-transition-htc.md:94]:
```suggestion
chtc_want_el8 = true
```

File: _uw-research-computing/os-transition-htc.md [hunk 10]
```diff
@@ -81,7 +87,13 @@ requirements = (Target.HasCHTCStaging == true)
 You can add the requirements for using CentOS Stream 8 like so:
 
 ``` {.submit}
-requirements = (Target.HasCHTCStaging == true) && (OpSysMajorVer == 7)
+requirements = (Target.HasCHTCStaging == true) && (OpSysMajorVer == 8)
+```
+Or use the +WantEL syntax:
```
Comment (rachellombardi) [inline _uw-research-computing/os-transition-htc.md:92]:
```suggestion
Or use the `chtc_want_el` syntax: 
```

File: _uw-research-computing/os-transition-htc.md [hunk 11]
```diff
@@ -40,33 +39,40 @@ version (or versions) for your jobs. This option is more limiting because
...
+# or 
+chtc_want_el8 = true
 ```
 
+To request that your jobs run on servers with CentOS 9 **only**, add one of the following requirements lines to your submit file: 
+
+```{.sub}
+requirements = (OpSysMajorVer == 9)
+# or
+chtc_want_el9 = true
```
Comment (rachellombardi) [inline _uw-research-computing/os-transition-htc.md:75]:
```suggestion
chtc_want_el9 = true 
```

```

File: _uw-research-computing/os-transition-htc.md [hunk 12]
```diff
@@ -40,33 +39,40 @@ version (or versions) for your jobs. This option is more limiting because
...
 ``` {.sub}
-requirements = (OpSysMajorVer == 7)
+requirements = (OpSysMajorVer == 8)
+# or 
+chtc_want_el8 = true
 ```
 
+To request that your jobs run on servers with CentOS 9 **only**, add one of the following requirements lines to your submit file: 
+
+```{.sub}
```
Comment (ChristinaLK) [inline _uw-research-computing/os-transition-htc.md:72]:
```suggestion
``` {.sub}
```

File: _uw-research-computing/os-transition-htc.md [hunk 13]
```diff
@@ -81,7 +87,13 @@ requirements = (Target.HasCHTCStaging == true)
 You can add the requirements for using CentOS Stream 8 like so:
 
 ``` {.submit}
```
Comment (ChristinaLK) [inline _uw-research-computing/os-transition-htc.md:89]:
```suggestion
``` {.sub}
```

**Review by ChristinaLK — APPROVED** [review]

**Pull request merged**